src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -412,6 +475,12 @@ async def create_pex(
         argv.extend(request.platforms.generate_pex_arg_list())
     else:
         argv.extend(request.interpreter_constraints.generate_pex_arg_list())
```

Comment:
This is the bug. I asserted in some comment or commit message that, in the case of internal_only we could leave in ICs for documentation. They do more and cause all interpreters >= 3.6 to be used (this 3.7). So the fix is to only add ICs if not internal_only.

Reply:
That fixed it! Thanks John!